src/DataProvider/SerializerAwareDataProviderTrait.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\DataProvider;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Inject serializer in data providers.
+ *
+ * @author Vincent Chalamon <vincent@les-tilleuls.coop>
+ */
+trait SerializerAwareDataProviderTrait
```

Comment:
I'm thinking of something about this trait,

Couln't we add the serialize and deserialize method directly in it ?

Since it's already some king of a proxy, that would avoid a function call to getSereliazer() IMO.